stac_api/api/extensions/tiles_search.py

```diff
@@ -0,0 +1,231 @@
+            "collections",
+            "ids",
+            "datetime",
+            "limit",
+            "query",
+            "token",
+            "fields",
+            "sortby",
+        ]
+        kwargs = {k: v for k, v in kwargs.items() if k not in remove_kwargs}
+
+        def _reader(asset: str, x: int, y: int, z: int, **kwargs: Any) -> ImageData:
+            # Take out reader_options for now because that's how I'm passing the
+            # postgres client to self.get_assets
+            # with self.reader(asset, **self.reader_options) as src_dst:
+            with self.reader(asset) as src_dst:
+                return src_dst.tile(x, y, z, **kwargs)
+
+        return mosaic_reader(mosaic_assets, _reader, x, y, z, **kwargs)
+
+
+@dataclass
+class AssetParams(DefaultDependency):
+    """Band names and Expression parameters."""
+
+    assets: Optional[str] = Query(
+        None,
+        title="Asset names",
+        description="comma-delimited asset names.",
+    )
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.assets is not None:
+            self.kwargs["assets"] = self.assets.split(",")
+
+
+@dataclass
+class DatasetDependency(DefaultDependency):
+    """Add STAC search params to query."""
+
+    collections: Optional[str] = Query(
+        None, title="Collections", description="Collections to search within"
+    )
+
+    ids: Optional[str] = Query(None)
+    datetime: Optional[str] = Query(None)
+    limit: Optional[int] = Query(10)
+    query: Optional[str] = Query(None)
+    token: Optional[str] = Query(None)
+    fields: Optional[str] = Query(None)
+    sortby: Optional[str] = Query(None)
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.collections:
+            self.kwargs["collections"] = self.collections.split(",")
+        if self.ids:
+            self.kwargs["ids"] = self.ids.split(",")
+        if self.datetime:
+            self.kwargs["datetime"] = self.datetime
+        if self.limit:
+            self.kwargs["limit"] = self.limit
+        if self.query:
+            self.kwargs["query"] = self.query
+        if self.token:
+            self.kwargs["token"] = self.token
+        if self.fields:
+            self.kwargs["fields"] = self.fields.split(",")
+        if self.sortby:
+            self.kwargs["sortby"] = self.sortby.split(",")
+
+
+@attr.s
+class TilesSearchExtension(ApiExtension):
+    """Custom Titiler Extension.
+
+    Custom extension for sending query results to a titiler DynamicSTAC endpoint
+    """
+
+    client: CoreCrudClient = attr.ib(default=attr.Factory(CoreCrudClient))
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        dynamic_stac_endpoint = MosaicTilerFactory(
+            reader=DynamicStacBackend,
+            reader_options={"client": self.client},
```

Comment:
What's the best way to pass the `client` to the `MosaicTilerFactory`? I need to access the client in the `MosaicBackend.get_assets` method.

Reply:
> but it seems like it shouldn't be on the reader_options.

I think it's ok to use the reader_options here and just pass a `backend_options` dict within. 

```python
reader_options={"backend_options":{"client": self.client}},
```

that's how we do for STAC https://github.com/developmentseed/cogeo-mosaic/blob/master/cogeo_mosaic/backends/stac.py#L76

I agree it's not perfect

Reply:
Wait, in fact you are already creating a custom Mosaic Backend so you can still use `reader_options={"client": self.client}` and make client a required parameter for your `DynamicStacBackend` backend